### ISSUE TITLE
Question about  the results

### ISSUE THREAD

**Issue by DongXiang-CV** [EdisonLeeeee/MaskGAE#2]
Hi authors, thank you for your great work. 

I have some questions about the results. I cloned your code and directly run it without any modification but I got the different results. For instance,  

`python train_nodeclas.py --dataset Cora --bn --l2_normalize --encoder_layers 3 --nodeclas_weight_decay 1e-4 --alpha 0.004 --mask Edge --lr 0.02`, I got the following results:
Start Training (Node Classification)...
Run 1: Best test accuray 81.20%.
Run 2: Best test accuray 81.20%.
Run 3: Best test accuray 81.00%.
Run 4: Best test accuray 81.00%.
Run 5: Best test accuray 81.20%.
Run 6: Best test accuray 81.10%.
Run 7: Best test accuray 81.10%.
Run 8: Best test accuray 81.20%.
Run 9: Best test accuray 81.40%.
Run 10: Best test accuray 81.40%.
##### Final Testing result (Node Classification)
All runs:
Highest Valid: 80.28 ± 0.23
   Final Test: 81.18 ± 0.14
(MaskGAE) 

This is different from the following results you have shown:

Start Training (Node Classification)...
Run 1: Best test accuray 83.50%.
Run 2: Best test accuray 83.60%.
Run 3: Best test accuray 83.70%.
Run 4: Best test accuray 83.50%.
Run 5: Best test accuray 83.60%.
Run 6: Best test accuray 83.60%.
Run 7: Best test accuray 83.90%.
Run 8: Best test accuray 83.60%.
Run 9: Best test accuray 83.40%.
Run 10: Best test accuray 83.70%.
##### Final Testing result (Node Classification)
All runs:
Highest Valid: 80.48 ± 0.23
   Final Test: 83.61 ± 0.14

`python train_nodeclas.py --dataset Citeseer --bn --l2_normalize --nodeclas_weight_decay 1 --alpha 0.008 --decoder_dropout 0.3`, I got the following results:

All runs:
Highest Valid: 97.24 ± nan
   Final Test: 96.75 ± nan
AP
All runs:
Highest Valid: 97.33 ± nan
   Final Test: 96.74 ± nan
Start Training (Node Classification)...
Run 1: Best test accuray 70.70%.
Run 2: Best test accuray 70.50%.
Run 3: Best test accuray 70.40%.
Run 4: Best test accuray 71.10%.
Run 5: Best test accuray 70.70%.
Run 6: Best test accuray 69.30%.
Run 7: Best test accuray 71.10%.
Run 8: Best test accuray 70.50%.
Run 9: Best test accuray 70.50%.
Run 10: Best test accuray 70.20%.
##### Final Testing result (Node Classification)
All runs:
Highest Valid: 71.58 ± 0.38
   Final Test: 70.50 ± 0.51
(MaskGAE) 

This is also different from the following results you have shown:

All runs:
Highest Valid: 97.73 ± nan
   Final Test: 96.42 ± nan
AP
All runs:
Highest Valid: 97.76 ± nan
   Final Test: 96.38 ± nan
Start Training (Node Classification)
Run 1: Best test accuray 74.60%.
Run 2: Best test accuray 73.40%.
Run 3: Best test accuray 73.90%.
Run 4: Best test accuray 72.70%.
Run 5: Best test accuray 73.40%.
Run 6: Best test accuray 73.80%.
Run 7: Best test accuray 73.00%.
Run 8: Best test accuray 74.00%.
Run 9: Best test accuray 73.30%.
Run 10: Best test accuray 72.80%.
Final Testing result (Node Classification)
All runs:
Highest Valid: 72.62 ± 0.36
   Final Test: 73.49 ± 0.59

I also find all examples given by nodeclas.ipynb can not match the given results. 

I do not know why and could you help me find the reasons？

Thank you very much!

**Comment by EdisonLeeeee** [comment]
That's weird. Could you please specify the version of the following packages you are using?
+ torch_geometric
+ torch_scatter
+ torch_sparse
+ torch_cluster
+ torch

**Comment by DongXiang-CV** [comment]
Hi, the packages I used are:

torch                              1.9.0+cu111
torch-cluster                      1.5.9
torch-geometric                2.0.4
torch-scatter                      2.0.9
torch-sparse                      0.6.10

**Comment by EdisonLeeeee** [comment]
I found the same issue when testing on another machine. This is mainly due to the difference in random seeds on different machines, and hence the optimal hyperparameters would be different. Could you try the following commands?
+ Cora
```bash
python train_nodeclas.py --dataset Cora --bn --l2_normalize --encoder_layers 2 --nodeclas_weight_decay 1e-4 --alpha 0.001 --lr 0.01

python train_nodeclas.py --dataset Cora --bn --l2_normalize --encoder_layers 2 --nodeclas_weight_decay 1e-4 --alpha 0.004 --mask Edge --lr 0.02
```


+ Citeseer
```python
python train_nodeclas.py --dataset Citeseer --bn --l2_normalize --nodeclas_weight_decay 5e-1 --alpha 0.001 --decoder_dropout 0.3 --lr 0.02

python train_nodeclas.py --dataset Citeseer --bn --l2_normalize --nodeclas_weight_decay 5e-1 --alpha 0.001 --decoder_dropout 0.3 --lr 0.02 --mask Edge --decoder_layers 3

```

Sorry for this, I'll investigate it further.

**Comment by zzzsh2000** [comment]
Hi authors, thank you for your great work.

I have some questions about the results. I cloned your code and directly run it without any modification but I got the different results. For instance,

python train_nodeclas.py --dataset Cora --bn --l2_normalize --encoder_layers 2 --nodeclas_weight_decay 1e-4 --alpha 0.001 --lr 0.01

Start Training (Node Classification)...
Run 1: Best test accuray 68.00%.
Run 2: Best test accuray 67.40%.
Run 3: Best test accuray 66.90%.
Run 4: Best test accuray 66.50%.
Run 5: Best test accuray 67.00%.
Run 6: Best test accuray 67.70%.
Run 7: Best test accuray 67.40%.
Run 8: Best test accuray 67.90%.
Run 9: Best test accuray 66.80%.
Run 10: Best test accuray 68.30%.
##### Final Testing result (Node Classification)
All runs:
Highest Valid: 66.00 ± 0.76
   Final Test: 67.39 ± 0.59

I do not know why and could you help me find the reasons？

Thank you very much!

**Comment by EdisonLeeeee** [comment]
Which versions of your PyTorch and PyG? The results look weird. Could you show me other results on large graphs such as Pubmed?

**Comment by zzzsh2000** [comment]
PyTorch  == 1.8.0
PyG  == 2.0.1

python train_nodeclas.py --dataset Pubmed --bn --l2_normalize --alpha 0.001  --encoder_dropout 0.5 --decoder_dropout 0.5

Start Training (Node Classification)...
Run 1: Best test accuray 67.90%.
Run 2: Best test accuray 66.30%.
Run 3: Best test accuray 67.70%.
Run 4: Best test accuray 67.30%.
Run 5: Best test accuray 67.60%.
Run 6: Best test accuray 68.40%.
Run 7: Best test accuray 67.00%.
Run 8: Best test accuray 67.90%.
Run 9: Best test accuray 68.00%.
Run 10: Best test accuray 67.20%.
##### Final Testing result (Node Classification)
All runs:
Highest Valid: 70.68 ± 0.69
   Final Test: 67.53 ± 0.60

**Comment by EdisonLeeeee** [comment]
While we have observed that performance fluctuations can occur in the Cora, CiteSeer, and PubMed citation datasets due to various factors such as environment, seed, and version, we have not observed such significant fluctuations. 

I am unable to reproduce your results and have no idea as to why that might be the case. Sorry about that. You may consider upgrading your PyTorch and PyG versions to see if that resolves the issue.
